reframe_translator.go

```diff
@@ -71,9 +72,11 @@ func (x *ReframeTranslatorService) FindProviders(ctx context.Context, key cid.Ci
 				outMsg.Err = fmt.Errorf("unexpected number of multihashes: %d", len(parsed.MultihashResults))
 				goto output
 			}
+			seen := make(map[peer.ID]struct{})
 			for _, prov := range parsed.MultihashResults[0].ProviderResults {
-				if isBitswapMetadata(prov.Metadata) {
+				if _, ok := seen[prov.Provider.ID]; isBitswapMetadata(prov.Metadata) && !ok {
```

Comment:
Check for existence first to save on unnecessary metadata decode.

```suggestion
				if _, ok := seen[prov.Provider.ID]; !ok && isBitswapMetadata(prov.Metadata) {
```